projects/ngx-charts-on-fhir/src/lib/fhir-chart/range-selector/range-selector.component.spec.ts

```diff
@@ -141,18 +154,28 @@ describe('RangeSelectorComponent', () => {
   });
 
   it('should check dateChange selected event for start date', () => {
-    const date: any = { value : new Date(2020, 2, 2) };
-    component.dateChange(date , 'min');
+    const date: any = { value: new Date(2020, 2, 2) };
+    component.dateChange(date, 'min');
     fixture.detectChanges();
     expect(component.minDate).toEqual(date.value);
   });
 
 
   it('should check dateChange selected event for end date', () => {
-    const date: any = { value : new Date(2020, 2, 2) };
-    component.dateChange(date , 'max');
+    const date: any = { value: new Date(2020, 2, 2) };
+    component.dateChange(date, 'max');
     fixture.detectChanges();
     expect(component.maxDate).toEqual(date.value);
   });
 
+  it('should check maxDate and minDate', async () => {
+    component.getMaxDateFromLayers(component.layers)
+    const componentMindate = new Date(component.minDate).getTime()
+    const componentMaxdate = new Date(component.maxDate).getTime()
+    if (component.layers) {
+      expect(componentMaxdate).toEqual(component.layers[0].datasets[0].data[1].x);
+      expect(componentMindate).toEqual(component.layers[2].datasets[0].data[1].x);
+    }
```

Comment:
This test will pass if component.layers is undefined. I don't think that's what we want. 
Try this instead:

```suggestion
    expect(componentMaxdate).toEqual(component.layers?.[0].datasets[0].data[1].x);
    expect(componentMindate).toEqual(component.layers?.[2].datasets[0].data[1].x);
```

Reply:
...although that will pass if both the layers and maxDate are undefined. We should probably use a literal date for expected value instead of pulling it from the component. Something like this:
```
expect(componentMaxdate).toEqual(new Date(1573505702))
```

Reply:
@semicolin here we need to typecast that number for compile 
![image](https://user-images.githubusercontent.com/117890382/212088385-186071e0-d4d5-4542-bb90-c16a3c99c803.png)